adam/learner/object_recognizer.py

```diff
@@ -202,6 +203,34 @@ def match_objects(
         candidate_object_subgraphs = self.extract_candidate_objects(perception_graph)
 
         for candidate_object_graph in candidate_object_subgraphs:
+            root_node = self._get_root_object_perception(
```

Comment:
Rather than do this on every `candidate_object_graph`, can we move it outside the loop and just search the graph for the `GROUND_PERCEPTION` node?